src/main/java/net/snowflake/client/core/SfSqlArray.java

```diff
@@ -82,6 +89,13 @@ public ResultSet getResultSet(long index, int count, Map<String, Class<?>> map)
   public void free() throws SQLException {}
 
   public String getJsonString() throws SQLException {
+    if (text == null) {
+      text = buildJsonStringFromElements(elements);
+    }
+    return text;
+  }
+
+  private static String buildJsonStringFromElements(Object elements) throws SQLException {
     try {
       return SnowflakeUtil.mapJson(elements);
```

Comment:
I think this method is used for both JSON and ARROW data. For arrow structured types the text field will always be set so it won't be executed. Now I think that it may be worth to check for potential divergence in primitive types though